### PR TITLE
[Not to be merged] initial Italian patterns

### PR DESCRIPTION
**Pull request by LionbridgeCS**
Italian DateTime, Numbers and NumbersWithUnit YAML files reviewed and localized. 
Not to be merged yet, debugging the supporting code. 
Question: Encountering Expected<1> actual <0> when running the test cases.

### CODE REVIEW

File: Patterns/Italian/Italian-DateTime.yaml [hunk 1]
```diff
@@ -1,162 +1,206 @@
 ---
 TillRegex: !simpleRegex
-  def: (?<till>fino al|fino a|al|a|--|-|—|——|~)
-RangeConnectorRegex : !simpleRegex
-  def: (?<and>e|al|a|--|-|—|——)
+  def: (?<till>fino a(l(l[aoe'])?|gli|i)?|a(l(l[aoe'])?|gli|i)?|e (il?|l[aoe']|gli)|--|-|—|——|~)
```
Comment (tellarin) [inline Patterns/Italian/Italian-DateTime.yaml]: Is this version of the YAML file in sync with the more recent changes in English? Maybe new regexes or tweaks to them?

File: Patterns/Italian/Italian-Numbers.yaml [hunk 2]
```diff
@@ -1,41 +1,45 @@
 ---
 LangMarker: Ita
 #Integer Regex
+DigitsNumberRegex: !simpleRegex
+  def: \d|\d{1,3}(\.\d{3})
 RoundNumberIntegerRegex: !simpleRegex
-  def: (cento|mille|milione|milioni|miliardo|miliardi|trilione|trilioni)
+  def: (milion[ei]?|miliard[oi]?|bilion[ei]?|trilion[ei]?)
```
Comment (tellarin) [inline Patterns/Italian/Italian-Numbers.yaml]: No number term for multiples of 100?

File: Patterns/Italian/Italian-Numbers.yaml [hunk 3]
```diff
@@ -1,41 +1,45 @@
...
-  def: ^({NegativeNumberTermsRegex}\s+).*
+  def: ^{NegativeNumberTermsRegex}.*
   references: [ NegativeNumberTermsRegex ]
-SeparaIntRegex: !nestedRegex
-  def: ({SupportThousandsRegex}(\s+{SupportThousandsRegex})*(\s+{BelowThousandsRegex})?|{BelowThousandsRegex})
-  references: [ SupportThousandsRegex, BelowThousandsRegex ]
-AllIntRegex: !nestedRegex
-  def: ({SeparaIntRegex}|mille(\s+{BelowThousandsRegex})?)
-  references: [ SeparaIntRegex, BelowThousandsRegex ]
+SeparaIntRegex: !nestedRegex #0,1,...,1.000.000,2.000.000,...,999.999.000.000,...,
```
Comment (tellarin) [inline Patterns/Italian/Italian-Numbers.yaml]: Feel free to add such comments in the English YAML too if they improve readability and understanding of the patterns.

File: Patterns/Italian/Italian-Numbers.yaml [hunk 4]
```diff
@@ -409,47 +419,75 @@ RoundNumberMap: !dictionary
...
-# TODO: modify below regex according to the counterpart in English
-  entries:
-    '': ''
-RelativeReferenceRelativeToMap: !dictionary
+RelativeReferenceMap: !dictionary
   types: [ string, string ]
-# TODO: modify below regex according to the counterpart in English
   entries:
-    '': ''
+    ultimo: N
```
Comment (tellarin) [inline Patterns/Italian/Italian-Numbers.yaml]: The behaviour for ordinal.relative has changed in /master. Can you please sync with the latest behaviour in English?

**Comment by tellarin** [comment]
@LionbridgeCS when updating this PR, can you please break it into at least 3 separates ones (by pattern file)?
Also, when making a PR with YAML changes, we also need the corresponding C# code generated from transformation to make sure there is no build error.
Lastly, please try to make a PR for each type as soon as possible and enabling its corresponding tests.
For example, one PR for Numbers (YAML, C# code, and enabled specs passing), then one for Ordinal, then Percentage, etc. Same per type approach for Units, then for DateTime. This greatly helps review and also debugging of issues.
Thanks!

**Comment by tellarin** [comment]
@LionbridgeCS BTW, GitHub now has the concept of WIP (work in progress) PRs. You can use that in a PR to make sure it's not merged by accident.

**Comment by LionbridgeCS** [comment]
@tellarin Thank you for the feedback. 
Should we close this PR and make separate ones ?

**Comment by tellarin** [comment]
@LionbridgeCS Either closing this one or updating it to focus on just one entity are fine.

**Pull request closed without merging**